### PR TITLE
Create a packaging project

### PR DESCRIPTION
**Pull request by SingleAccretion**
Creates a packaging project using `nugetizer` infrastructure.

One question I have here is what to do with the `BuildingForSourceGenerator` little property. It may be possible that in the future you'll want to have a separate analyzer project in which case it would be prudent to put this logic `Directory.Build.targets` (it'll also make the project file look cleaner).

However, it does increase the complexity a bit and makes for one more moving piece (you cannot really determine what's up with the build by just looking at the project file anymore).

I'll defer judgement on this issue to you Sergio.

Also, the old `.nuspec` should probably be deleted?

### CODE REVIEW

**Comment by SingleAccretion** [comment]
Note that `global.json` is only really needed until the fix for BDN flows to `dotnet/installer` as it affects only people on the nightly SDK builds (like myself 😄). It can (and probably should) be safely taken out.

File: src/ComputeSharp.Core/ComputeSharp.Core.csproj [hunk 1]
```diff
@@ -4,11 +4,27 @@
...
+    <!--
+      This is needed so that the packaging for the source generator is correct
+      We want to have the package references of this project be private when packing the SG
+      (The nugetizer infrastructure recognizes that and packages the dependencies directly alongside the main binary)
+      And public dependencies when building for the main project
+      Right now this is conditioned on the target framework, so this property is primarily for readability
+      This may change in the future
+    -->
+    <BuildingForSourceGenerator Condition="'$(TargetFramework)' == 'netstandard2.0'">false</BuildingForSourceGenerator>
+    <BuildingForSourceGenerator Condition="'$(TargetFramework)' == 'netstandard2.0'">true</BuildingForSourceGenerator>
```
Comment (Sergio0694) [inline src/ComputeSharp.Core/ComputeSharp.Core.csproj:19]: I'm confused, was one of these two checks meant to be `!=`? 🤔

Comment (SingleAccretion) [reply src/ComputeSharp.Core/ComputeSharp.Core.csproj:19]: One of them was not meant to exist.

Comment (SingleAccretion) [reply src/ComputeSharp.Core/ComputeSharp.Core.csproj]: Pushed the fix.

**Comment by Sergio0694** [comment]
This looks amazing, thank you so much for the help! 😄

image

Looks great! And I love how the solution is super clean too! 🎉

**Pull request merged**